src/backends/backend/tritonbackend.h

```diff
@@ -0,0 +1,566 @@
+/// tensor object matches that of the request and so the input tensor
+/// object should not be accessed after the request object is
+/// released.
+///
+/// \param request The inference request.
+/// \param index The index of the input tensor. Must be 0 <= index <
+/// count, where count is the value returned by
+/// TRITONBACKEND_RequestInputCount.
+/// \param input Returns the input tensor corresponding to the index.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONBACKEND_EXPORT TRITONSERVER_Error* TRITONBACKEND_RequestInput(
+    TRITONBACKEND_Request* request, const uint32_t index,
+    TRITONBACKEND_Input** input);
+
+/// Get the number of output tensors specified in the request.
+///
+/// \param request The inference request.
+/// \param count Returns the number of output tensors.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONBACKEND_EXPORT TRITONSERVER_Error* TRITONBACKEND_RequestOutputCount(
+    TRITONBACKEND_Request* request, uint32_t* count);
+
+/// Get a requested output tensor. The lifetime of the returned output
+/// tensor object matches that of the request and so the output tensor
+/// object should not be accessed after the request object is
+/// released.
+///
+/// \param request The inference request.
+/// \param index The index of the output tensor. Must be 0 <= index <
+/// count, where count is the value returned by
+/// TRITONBACKEND_RequestOutputCount.
+/// \param output Returns the output tensor corresponding to the index.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONBACKEND_EXPORT TRITONSERVER_Error* TRITONBACKEND_RequestOutput(
+    TRITONBACKEND_Request* request, const uint32_t index,
+    TRITONBACKEND_RequestedOutput** output);
+
+/// Release the request. The request should be released when it is no
+/// longer needed by the backend. After this call returns the
+/// 'request' object is no longer valid and must not be used. Any
+/// tensor names, data types, shapes, etc. returned by
+/// TRITONBACKEND_Request* functions for this request is no longer
+/// valid. If a persistent copy of that data is required it must be
+/// created before calling this function.
+///
+/// \param request The inference request.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONBACKEND_EXPORT TRITONSERVER_Error* TRITONBACKEND_RequestRelease(
+    TRITONBACKEND_Request* request);
+
+///
+/// TRITONBACKEND_ResponseFactory
+///
+/// Object representing an inference response factory.
+///
+
+/// Create the response factory associated with a request. This
+/// function can be called only once for a given request. Subsequent
+/// calls with result in a TRITONSERVER_ERROR_ALREADY_EXISTS error.
+///
+/// \param factory Returns the new response factory.
+/// \param request The inference request.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONBACKEND_EXPORT TRITONSERVER_Error* TRITONBACKEND_ResponseFactoryNew(
```

Comment:
So the response factory is going to be decoupled from the request?

Reply:
Yes, the response factory allows you to send requests even after you have released the request itself. Releasing the request  also releases all the buffers holding the input tensors so you want to release the request as soon as possible. You can hold onto the response factory for much longer so that you can generate as many responses as needed.